pkg/tuf/client.go

```diff
@@ -313,10 +313,34 @@ func NewFromEnv(_ context.Context) (*TUF, error) {
 	}
 
 	// Initializes a new TUF object from the local cache or defaults.
-	return initializeTUF(mirror, nil, getEmbedded(), false)
+	return initializeTUFOnce(mirror, nil, getEmbedded(), false)
 }
 
+// Initialize creates a new TUF client, if one has not already been initialized,
+// and writes the cache to disk if requested. Thread safe.
 func Initialize(ctx context.Context, mirror string, root []byte) error {
+	// Initialize the client. Force an update with remote.
+	if _, err := initializeTUFOnce(mirror, root, getEmbedded(), true); err != nil {
+		return err
+	}
+
+	// Store the remote for later if we are caching.
+	if !noCache() {
+		remoteInfo := &remoteCache{Mirror: mirror}
+		b, err := json.Marshal(remoteInfo)
+		if err != nil {
+			return err
+		}
+		if err := os.WriteFile(cachedRemote(rootCacheDir()), b, 0o600); err != nil {
+			return fmt.Errorf("storing remote: %w", err)
+		}
+	}
+	return nil
+}
+
+// InitializeAlways creates a new TUF client. Repeated calls to Initialize
```

Comment:
nit, Repeated calls to InitializeAlways?
This now allows one to have multiple clients simultaneously which I was having a bit of a rub before. I think by not forcing a singleton is nice, but the one thing that might be problematic still is the use of `rootCacheDir` below since now different clients will have a shared state in the form of the local cache, which is problematic. I think if you allow the caller to specify the cachedir here would be better.
And while you're at it, I think rather than having the getEmbedded live inside the function, taking in the fs.FS as an argument would be better, and would probably make the testing easier so one doesn't have to override the function (which IIRC caused some hilarity in the past as well. If not here, surely elsewhere :) ).

Reply:
>  figure those should be hoisted to sigstore TUF client (something like GetFulcioRoots) since that's what (I think at least) the sigstore TUF client is adding on top of raw TUF, semantic meaning to various keys.

YES! I would have liked an API like:

````
GetTufObj(TufParams) -> TufObj

TufObj.GetFulcio -> Fulcio
```